Deprecate redundant `fract_delta` property in GPUParticles

Calinou opened the pull request:
This property was made redundant by the addition of interpolation in GPUParticles. It's now always enabled.

CPUParticles still features a `fract_delta` property for now. Should we remove it as well?

See [discussion on the Godot Contributors Chat](https://chat.godotengine.org/channel/rendering?msg=G8cNTJHiDcnEwQbuy).

lawnjelly commented:
> CPUParticles still features a fract_delta property for now. Should we remove it as well?

Note that #52846 now uses the `fract_delta` for the timing calculations for "streaking" in interpolated CPUParticles (in 3.x but will likely be ported to 4.x). Streaking is using the `fract_delta` to back-calculate a "primed" position for restarting particle so they appear on an infinite timeline, rather than bunched up at update times only.

I haven't yet investigated how the GPUParticles work or whether they could also do streaking (I'm sure it would be welcome there too if we can do it, although it doesn't have to be part of the interpolation PR). I don't know yet how the interpolation works in 4.x, whether it already can do this.

MJacred commented:
@lawnjelly: fixed timestep interpolation for Godot 4 was implemented (#92391, #91818). A code search on Github says that  `fract_delta` is unused.  
Is there still sth. missing for Godot 4? If not, this property seems to remain redundant and could therefore be removed

Calinou commented:
> A code search on Github says that fract_delta is unused.

`fractional_delta` (and its resulting shader constant `USE_FRACTIONAL_DELTA`) are still used though. `fract_delta` is only used in the property names; all methods use `fractional_delta` instead.

I still think the property could be removed as I don't see any use cases to disabling it (unlike interpolation, which you may want to disable to sidestep rendering issues or have a retro look).

Calinou committed:
Deprecate redundant `fract_delta` property in GPUParticles

This property was made redundant by the addition of interpolation
in GPUParticles. It's now always enabled and hidden in the inspector.

CPUParticles' `fract_delta` property is still effective after this change.
